multimodal/src/autogluon/multimodal/data/process_document.py

```diff
@@ -220,6 +220,35 @@ def apply_ocr(self, doc_image):
         results[BBOX] = boxes
         return results
 
+    def get_ocr_features(self, doc_path, doc_image):
+        """
+        Get the OCR features, and avoid running OCR multiple times.
+
+        Parameters
+        ----------
+        doc_path
+            The document path.
+        doc_image
+            The document image.
+
+        Returns
+        -------
+        Extracted words, image pixel features, and bounding boxes.
+        """
+        # The OCR process is time-consuming, so apply OCR on each image only once.
+        if doc_path not in self.documents:
+            ocr_res = self.apply_ocr(doc_image)
+            # store the ocr results.
+            self.documents.update({doc_path: ocr_res})
```

Comment:
Do we need to empty `self.documents` before saving the processor?

Reply:
good suggestion, will handle that